core/services/relay/evm/evm.go

```diff
@@ -410,7 +408,6 @@ func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, tra
 		checker.CheckerType = txm.TransmitCheckerTypeSimulate
 	}
 
-	gasLimit := configWatcher.chain.Config().EVM().GasEstimator().LimitDefault()
```

Comment:
I see this is being passed from three call sites. Why not keep this as is and add logic for pluginGasLimit override (same logic as lines 464-471)